src/ServicePulse.Host/app/js/services/services.monitoring.js

```diff
@@ -29,7 +42,24 @@
             mappedUrls.forEach(function (url) {
                 $http.get(url)
                     .then(function (result) {
+                        if (connectionToasts[url] !== undefined) {
+                            var message = connectionToasts[url];
+                            // doesn't exist
+                            toastr.clear(message);
+                            connectionToasts[url] = undefined;
+                        }
                         observer.onNext(result.data);
+                    }, function (error) {
```

Comment:
so there are other options which I ruled out:
* call OnError on the endpoints stream: Worst approach. Will cause all subscribers to unsubscribe and will require "reconnect" logic on every subscriber. Additionally it will require a hold time on every subscriber to ensure we're not pulling constantly in case of an error. This massively complicates the rx code and I can tell you from my previous rx experience, you don't ever want to have OnError on a stream which should just reliably push data to several subscribers repeatedly.
* Make the pushed data on the stream an `Either` value which provides either the retrieved data or the error so subscribers can be aware of an error happening. This would work well in case you want to show a view specific error but it doesn't align with the concept of toast notifications which are not view-bound (which subscriber is responsible of raising the notification?). You would require a dedicated subscriber raising the notification which is tricky, because you don't want the notification logic to subscribe the same way as a view because it would start polling data even when no view is interested in data. This approach would work best with view-specific error messages.
* to tackle the last problem, you would have to provide a dedicated error-stream as a hot observable which is completely detached from the actual endpoints data stream for the component which subscribers to service failures and will display them in a view-agnostic way. I think this approach would align best with the current design and the fact we're using toast-notifications.

Given that I was not sure whether we will stick with the toastr notifications or not (view specific logic provides much more flexibility and visualization options), I didn't want to make the choice between option 2/3 and they heavily depend on this design decision.